torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -418,6 +418,17 @@ const auto kLoopbackAddress = "127.0.0.1";
 
 } // namespace
 
+std::vector<at::Tensor> ProcessGroupGloo::AsyncWork::result() {
+  TORCH_CHECK(
+      isCompleted(),
+      "Work needs to be completed before calling result(). "
+      "Should call wait() before result().");
+  TORCH_CHECK(
+      outputTensors_.size() <= 1,
+      "work result does not support list of list");
+  return outputTensors_.size() == 0 ? std::vector<at::Tensor>() : outputTensors_[0];
```

Comment:
nit: std::vector<at::Tensor>() can be replaced by "{}"

Reply:
c++ cannot figure out the types in ternary expression, so it complains about {} symbol here

Reply:
Interesting to know this, because ideally the compiler should be able to figure it out as you have specified the return type in the function signature.

Out of curiosity, if you do this instead:
`outputTensors_.size() == 1 ? outputTensors_[0] : {};`
Will the compiler be able to figure it out?

I am not saying that you need to change your code though.